modules/mod_ginger_rdf/controllers/controller_hydra.erl

```diff
@@ -76,8 +78,8 @@ to_hydra(ReqData, State) ->
         total = Total,
         next = Next,
         prev = Previous
-    } = z_search:search_pager({query, [{query_id, QueryId}]}, Page, Limit, Context),
-    RdfResults = [m_rdf_export:to_rdf(Id, [State#state.ontology], Context) || Id <- Result],
+    } = z_search:search_pager({State#state.query_type, search_arguments(QueryId, Context)}, Page, Limit, Context),
```

Comment:
I wonder if the `query_type` should come from the resource - is it a parameter of the hydra request (similar to the ontology parameter)? I think the query type is inseparable from the collection resource because it could result in a completely different collection of data if the type is changed.

Reply:
The `query_type` comes from the dispatch rule (just like `ontologies` does). 

> I think the query type is inseparable from the collection resource because it could result in a completely different collection of data if the type is changed.

True, but we have (e.g.) `ginger_search` for Ginger-specific search criteria, which may be used in the query. Anyway, I could hard-code this to `query` but decided that parameterising in the dispatch rule is a bit better.